### PR TITLE
Fix misc issues and enable v8r64 FPU

### PR DESCRIPTION
**Pull request by povergoing**
This PR fixes some issues and enables the FPU for v8R AArch64

- Fix text exception by increasing the stack size
- Remove LOG print before MPU is enabled to avoid the potential un-alignment instruction called before MPU is enabled.
- Fix the STACK_INIT during the reset
- Add cpu_map to have a correct mapping of CPU id and MPID
- ~~Fix a deadlock case when using FPU in the spin-locked section (**this workaround fixes modified the spinlock, RFC**)
        Using FPU with FPU sharing in a spin-locked critical section might cause a 'deadlock'. When core 0 got a spinlock that other cores are waiting for, it will 'deadlock' if the core 0 current threads' FPU context is living in another one that is waiting for the spinlock but never receives the FPU IPI. CC @npitre~~ https://github.com/zephyrproject-rtos/zephyr/pull/58086 fixed the deadlock issue

- Enable FPU and FPU_SHARING for v8r aarch64

Tested on all runnable Arm64 platforms including `fvp_baser_aemv8r` and `fvp_baser_aemv8r_smp`

### CODE REVIEW

File: include/zephyr/spinlock.h [hunk 1]
```diff
@@ -153,6 +153,10 @@ static ALWAYS_INLINE k_spinlock_key_t k_spin_lock(struct k_spinlock *l)
 
 #ifdef CONFIG_SMP
 	while (!atomic_cas(&l->locked, 0, 1)) {
+#if defined(CONFIG_ARM64) && defined(CONFIG_FPU_SHARING)
+		extern void z_arm64_flush_local_fpu_proactive(void);
+		z_arm64_flush_local_fpu_proactive();
+#endif
```
Comment (carlocaione) [inline include/zephyr/spinlock.h:159]: uhm, this could raise some eyebrows. Can you detach this from this PR and make a standalone PR only for this?

Comment (povergoing) [reply include/zephyr/spinlock.h:159]: Yes I will, and I also expect there is a better way to fix the issue

**Comment by npitre** [comment]
@povergoing: The deadlock is a very interesting one.

**Comment by povergoing** [comment]
I think this is ready for review :)

**Review by npitre — APPROVED** [review]

**Review by SgrrZhf — APPROVED** [review]

**Review by carlocaione — APPROVED** [review]
